src/os/bluestore/BlueStore.cc

```diff
@@ -18494,6 +18580,14 @@ int calc_allocator_image_trailer_size()
 //-----------------------------------------------------------------------------------
 int BlueStore::__restore_allocator(Allocator* allocator, uint64_t *num, uint64_t *bytes)
 {
+  if (cct->_conf->bluestore_debug_inject_allocation_from_file_failure > 0) {
+     boost::mt11213b rng(time(NULL));
+    boost::uniform_real<> ur(0, 1);
+    if (ur(rng) < cct->_conf->bluestore_debug_inject_allocation_from_file_failure) {
```

Comment:
I think <= is "almost" better, as it hits case when conf=1 and rng()=1

Reply:
uniform_real returns a random floating-point value distributed in the range [min..max). Hence your case isn't valid - rng() is never equal to 1